docs/national-document-repository-api-spec.md

```diff
@@ -0,0 +1,626 @@
+# NHS National Document Repository (NDR) API Specification
+
+## Overview
+
+This document provides a comprehensive specification of all Lambda functions and API endpoints in the NHS National Document Repository system. The NDR is a cloud-based service for managing patient documents within the NHS.
+
+## Authentication
+
+Most endpoints require custom authorization via the `repo_authoriser` Lambda function. Some endpoints use API keys for machine-to-machine communication.
+
+---
+
+## API Endpoints
+
+### - Patient Search
+
+**Handler**: [`search_patient_details_handler.lambda_handler`](./national-document-repository/lambdas/handlers/search_patient_details_handler.py)
+
+**Endpoint**: `/SearchPatient`  
```

Comment:
Probably not for this ticket, but should we be putting all these api specs in "OpenApi Spec/Swagger"